### ISSUE TITLE
Saved weight is too large on training

### ISSUE THREAD

**Issue by Mistsink** [ultralytics/ultralytics#16586]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I train the yolo11x.pt model, the weight file saved on the end of the epoch is too large. The original yolo11x.pt only 110M, I mean the file size. But the file in the runs/detect/train/weight is 218M, all the best.pt and last.pt. Now the training loop has not done. I am worried that something might be wrong. Otherwise, why is the file saved after training twice as large?

The code: 
```
from ultralytics import YOLO
model = YOLO("yolo11x.pt")

epoch = 300
model.train(
    data="./datasets/filter/data.yaml",
    epochs=epoch,
    workers=4,
    device=[0, 1, 2, 3],
    imgsz=640,
    close_mosaic=epoch,
    batch=32,)
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Mistsink, thanks for reaching out and for your interest in Ultralytics 🚀!

For new users, our [Documentation](https://docs.ultralytics.com) is a fantastic resource. You'll find various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might provide some clarity on common inquiries.

If this is a 🐛 Bug Report, we request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist in diagnosing the issue more effectively. Providing additional details such as log outputs or errors may also help.

If your question pertains to custom training ❓, please include as much context as possible—such as dataset examples and training logs. Make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our lively Ultralytics community on [Discord](https://ultralytics.com/discord) 🎧 for real-time discussions, explore [Discourse](https://community.ultralytics.com) for more in-depth topics, or connect with others on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure you are on the latest version, update the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 functionality is verified in several environments with pre-installed dependencies, such as:

- **Notebooks** with GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VMs: [GCP Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** AMIs: [AWS Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge updates based on our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests. It confirms correct YOLOv8 functionality on major platforms every 24 hours.

This is an automated response, and an Ultralytics engineer will review your issue soon. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
During training, the checkpoint also contains the optimizer state. This gets removed once the training finishes. So you should just let it run until the end.

**Comment by Mistsink** [comment]
Thank you very much, I will check the size on the end.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
